benchmark/src/main/java/test/CacheReadBenchmark.java

```diff
@@ -0,0 +1,75 @@
+package test;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+
+import org.openjdk.jmh.annotations.*;
+
+import com.google.common.base.Optional;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import io.servicefabric.transport.utils.memoization.Computable;
+import io.servicefabric.transport.utils.memoization.ConcurrentMapMemoizer;
+
+@Fork(2)
+@State(Scope.Thread)
+@Threads(4)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class CacheReadBenchmark {
+
+	static final String[] GENERIC_ARGS = new String[] {
+			"java.lang.String",
+			"java.lang.Long",
+			"java.lang.Integer",
+			"java.lang.Short",
+			"java.lang.Character",
+			"java.lang.Byte",
+			"io.servicefabric.transport.protocol.MessageSchema",
+	};
+
+	LoadingCache<String, Optional<Class>> guavaCache = CacheBuilder.newBuilder()
+			.expireAfterAccess(1, TimeUnit.HOURS)
```

Comment:
Remove `.expireAfterAccess(1, TimeUnit.HOURS)`, otherwise comparison is not fair.